sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -260,26 +260,50 @@ object IntervalUtils {
       try {
         units(i) match {
           case "year" =>
-            months = Math.addExact(months, Math.multiplyExact(values(i).toInt, 12))
+            months = new BigDecimal(values(i)).multiply(new BigDecimal(MONTHS_PER_YEAR))
```

Comment:
This looks like a new feature to me. Can we fix the bug first and then propose new features?

Reply:
+1 for @cloud-fan 's comment.

Reply:
OK, I will fix the bug first, a followup for [SPARK-29532]? or do we need a jira to track?

Reply:
Yes. A follow-up for the original JIRA.

Reply:
Sorry, I already create a new Jira in a hurry. also, this line may not blame to SPARK-29532

Reply:
I'd just make a follow-up PR to the other JIRA to fix the bug.